### PR TITLE
[SPARK-18870] Disallowed Distinct Aggregations on Streaming Datasets

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?

Check whether Aggregation operators on a streaming subplan have aggregate expressions with isDistinct = true.

## How was this patch tested?

Added unit test

### CODE REVIEW

**Comment by tdas** [comment]
@marmbrus

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala [hunk 1]
```diff
@@ -95,6 +96,15 @@ object UnsupportedOperationChecker {
       // Operations that cannot exists anywhere in a streaming plan
       subPlan match {
 
+        case Aggregate(_, aggregateExpressions, child) =>
+          val distinctAggExprs = aggregateExpressions.flatMap { expr =>
+            expr.collect { case ae: AggregateExpression if ae.isDistinct => ae }
+          }
+          throwErrorIf(
+            child.isStreaming && distinctAggExprs.nonEmpty,
+            "Distinct aggregations are not supported on streaming DataFrames/Datasets, unless" +
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala]: you need an extra space here. I'd also recommend users using approximate distinct.

**Comment by tdas** [comment]
jenkins test this please

**Comment by marmbrus** [comment]
LGTM

**Comment by tdas** [comment]
Merging to 2.1 and master

**Pull request closed without merging**